sdk/python/kfp/components/pipeline_task.py

```diff
@@ -563,8 +563,13 @@ def my_pipeline():
                 task1 = my_component(text='1st task')
                 task2 = my_component(text='2nd task').after(task1)
         """
+        from kfp.components.tasks_group import TasksGroupType
+
         for task in tasks:
-            if task.parent_task_group is not self.parent_task_group:
+            if task.parent_task_group is not self.parent_task_group and task.parent_task_group.group_type in [
+                    TasksGroupType.CONDITION, TasksGroupType.FOR_LOOP,
```

Comment:
I don't think the check based on group_type is the right fix here. I think it would still raise error with the following valid topology case, right?
```python
with dsl.Condition(...):
    task1 = component(...)
    with dsl.Condition(...):
        task2 = component(...)
        task2.after(task1)
```